engine.py

```diff
@@ -66,50 +61,163 @@ def create_shotgun_menu(self):
+        :param handler: Log handler that this message was dispatched from
+        :type handler: :class:`~python.logging.LogHandler`
+        :param record: Std python logging record
+        :type record: :class:`~python.logging.LogRecord`
+        """
+        log_debug = record.levelno < logging.INFO and sgtk.LogManager().global_debug
+        log_info_above = record.levelno >= logging.INFO
+        if log_debug or log_info_above:
+            msg = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S,%f")[:-3]
+            msg += " "
+            msg += handler.format(record)
+            print msg
 
     def _create_dialog(self, title, bundle, widget, parent):
-        from tank.platform.qt import QtCore
+        """
+        Create a TankQDialog with the specified widget embedded. This also connects to the
+        dialogs dialog_closed event so that it can clean up when the dialog is closed.
+        .. note:: For more information, see the documentation for :meth:`show_dialog()`.
+        :param title: The title of the window
+        :param bundle: The app, engine or framework object that is associated with this window
+        :param widget: A QWidget instance to be embedded in the newly created dialog.
+        :type widget: :class:`PySide.QtGui.QWidget`
+        """
+        from sgtk.platform.qt import QtCore
         dialog = super(TDE4Engine, self)._create_dialog(title, bundle, widget, parent)
         dialog.setWindowFlags(dialog.windowFlags() | QtCore.Qt.WindowStaysOnTopHint)
         dialog.setWindowState(
             (dialog.windowState() & ~QtCore.Qt.WindowMinimized) | QtCore.Qt.WindowActive)
         dialog.raise_()
         dialog.activateWindow()
         return dialog
+
+    #########################################################################################
+    # callbacks
+
+    def _jump_to_shotgun(self):
+        """
+        Jump to shotgun, launch web browser
+        """
+        from sgtk.platform.qt import QtCore, QtGui
+        url = self.context.shotgun_url
+        QtGui.QDesktopServices.openUrl(QtCore.QUrl(url))
+
+    def _jump_to_filesystem(self):
+        """
+        Jump from context to the filesystem
+        """
+        # launch one window for each location on disk
+        paths = self.context.filesystem_locations
+        for disk_location in paths:
+
+            # get the setting
+            system = sys.platform
+
+            # run the app
+            if system == "linux2":
+                cmd = 'xdg-open "%s"' % disk_location
+            elif system == "darwin":
+                cmd = 'open "%s"' % disk_location
+            elif system == "win32":
+                cmd = 'cmd.exe /C start "Folder" "%s"' % disk_location
+            else:
+                raise Exception("Platform '%s' is not supported." % system)
+
+            exit_code = os.system(cmd)
+            if exit_code != 0:
+                self.logger.error("Failed to launch '%s'!", cmd)
```

Comment:
I guess since we're adding a copy/our implementation of `_jump_to_fs`, might as well make the code better

Add this to top of file

```python
import subprocess
```

Then

```suggestion
    def _jump_to_fs(self):
        """
        Jump from context to FS
        """
        # launch one window for each location on disk
        paths = self.context.filesystem_locations

        # get the setting
        system = sys.platform

        # run the app
        if system == "linux2":
            cmd = ["xdg-open"]
        elif system == "darwin":
            cmd = ["open"]
        elif system == "win32":
            cmd = ["cmd.exe", "/C", "start", "Folder"]
        else:
            raise Exception("Platform '%s' is not supported." % system)

        for disk_location in paths:
            args = cmd + [disk_location]
            try:
                subprocess.check_call(args)
            except Exception as error:
                cmdline = subprocess.list2cmdline(args)
                self.logger.error("Failed to launch '%s'!", cmdline)
```